provider/schema/dialect.go

```diff
@@ -0,0 +1,305 @@
+		fmt.Sprintf("COMMENT ON COLUMN %s.%s IS '%s=%s.%s;'", t.Name, pc.Name, fkCommentKey, parent.Name, cqIdColumn.Name),
+	}
+}
+
+func (d TSDBDialect) DBTypeFromType(v ValueType) string {
+	return PostgresDialect{}.DBTypeFromType(v)
+}
+
+func (d TSDBDialect) GetResourceValues(r *Resource) ([]interface{}, error) {
+	return doResourceValues(d, r)
+}
+
+// GetFKFromComment gets a column comment and parses the parent table reference from it
+func GetFKFromComment(text string) (table string, column string) {
+	c := parseDialectComments(text)
+	v := c[fkCommentKey]
+	if v == "" {
+		return
+	}
+	tableCol := strings.SplitN(v, ".", 2)
+	if len(tableCol) != 2 {
+		return // invalid
+	}
+	return tableCol[0], tableCol[1]
+}
+
+func doResourceValues(dialect Dialect, r *Resource) ([]interface{}, error) {
+	values := make([]interface{}, 0)
+	for _, c := range dialect.Columns(r.table) {
+		v := r.Get(c.Name)
+		if err := c.ValidateType(v); err != nil {
+			return nil, err
+		}
+		if c.Type == TypeJSON {
+			if v == nil {
+				values = append(values, v)
+				continue
+			}
+			if reflect2.TypeOf(v).Kind() == reflect.Map {
+				values = append(values, v)
+				continue
+			}
+			switch data := v.(type) {
+			case map[string]interface{}:
+				values = append(values, data)
+			case string:
+				newV := make(map[string]interface{})
+				err := json.Unmarshal([]byte(data), &newV)
+				if err != nil {
+					return nil, err
+				}
+				values = append(values, newV)
+			case *string:
+				var newV interface{}
+				err := json.Unmarshal([]byte(*data), &newV)
+				if err != nil {
+					return nil, err
+				}
+				values = append(values, newV)
+			case []byte:
+				var newV interface{}
+				err := json.Unmarshal(data, &newV)
+				if err != nil {
+					return nil, err
+				}
+				values = append(values, newV)
+			default:
+				d, err := json.Marshal(data)
+				if err != nil {
+					return nil, err
+				}
+				var newV interface{}
+				err = json.Unmarshal(d, &newV)
+				if err != nil {
+					return nil, err
+				}
+				values = append(values, newV)
+			}
+		} else {
+			values = append(values, v)
+		}
+	}
+	return values, nil
+}
+
+func findParentIdColumn(t *Table) (ret *Column) {
+	for _, c := range t.Columns {
+		if c.Meta().Resolver != nil && c.Meta().Resolver.Name == "schema.ParentIdResolver" {
+			return &c
+		}
+	}
+
+	// Support old school columns instead of meta, this is backwards compatibility for providers using SDK prior v0.5.0
+	/*
+		for _, c := range t.Columns {
+			if strings.HasSuffix(c.Name, "cq_id") && c.Name != "cq_id" {
+				return &c
+			}
+		}
+	*/
+
+	return nil
+}
+
+func parseDialectComments(text string) map[string]string {
+	// format is key=value;...
+
+	parts := strings.Split(text, ";")
+	ret := make(map[string]string, len(parts))
+	for i := range parts {
+		kv := strings.SplitN(parts[i], "=", 2)
+		if len(kv) == 1 {
+			ret[kv[0]] = ""
+		} else {
+			ret[kv[0]] = kv[1]
+		}
+	}
+	return ret
+}
```

Comment:
let's sync on this. could keep these in a separate "FK list" metatable somewhere, or tweak the history setup routines to work on schema.Tables rather than the whole db.